keys/key.go

```diff
@@ -38,11 +41,22 @@ func LoadRSAPrivate(privKeyPath string) (*rsa.PrivateKey, error) {
 	if block == nil {
 		return nil, errors.New(ErrMsgPrivKeyNotPEMEncoded)
 	}
-	// check if it is a RSA PRIVATE KEY
-	if got, want := block.Type, "RSA PRIVATE KEY"; got != want {
+
+	switch block.Type {
+	case blockTypePKCS1:
+		return x509.ParsePKCS1PrivateKey(block.Bytes)
+	case blockTypePKCS8:
+		key, err := x509.ParsePKCS8PrivateKey(block.Bytes)
+		if err != nil {
+			return nil, err
+		} else if rsaKey, ok := key.(*rsa.PrivateKey); !ok {
+			return nil, errors.Errorf("key type '%T' not supported", key)
```

Comment:
nit, but maybe `*key` here would be better?

Reply:
also: is this line covered by unit tests?

Reply:
Removed the formatting directive as it may be misleading.